### DISCUSSION TITLE
Being able to downgrade to a previous CasaOS release if nessesary.

### DISCUSSION THREAD

**Discussion by MitasTech** [IceWhaleTech/CasaOS#289] [Ideas]
I'm on version 3.1.1 and i have about 2.5TB of data on my server. I'm a bit afraid to update to the new update just in case something goes bad because it happened in the past. I really want to update to get the new features and by having this feature would make a lot of people feel safer when updating and also help anyone that actually face issues. 
This is just my opinion of course, but it would be great to have a feature that allows downgrading.

**Comment by AuthorShin** [comment]
Exactly, it's necessary.
I have casaOS on three device and one is at `v0.3` which is the good one that I really like and other two are `v0.3.2` that I found very buggy and troublesome. I there was an option for downgrading I would definitely migrate to `v0.3`.

**Comment by NUR-E-ELAHI** [comment]
I think its necessary too! 
Im also  facing some issues on the latest update of 0.3.2 and  want to downgrade!
